### PR TITLE
SILOptimizer: fix a stack nesting problem when inlining coroutines

### PR DESCRIPTION
**Pull request by eeckstein**
Beside fixing the compiler crash, this change also improves the stack-nesting correction mechanisms in the inliners:

* Instead of trying to correct the nesting after each inlining of a callee, correct the nesting once when inlining is finished for a caller function.
This fixes a potential compile time problem, because StackNesting iterates over the whole function.
In worst case this can lead to quadratic behavior in case many begin_apply instructions with overlapping stack locations are inlined.

* Because we are doing it only once for a caller, we can remove the complex logic for checking if it is necessary.
We can just do it unconditionally in case any coroutine gets inlined.
The inliners iterate over all instruction of a function anyway, so this does not increase the computational complexity (StackNesting is roughly linear with the number of instructions).

rdar://problem/47615442

### CODE REVIEW

File: include/swift/SILOptimizer/Utils/SILInliner.h [hunk 1]
```diff
@@ -67,6 +67,17 @@ class SILInliner {
   /// attempts to inline \arg AI and this returns false, an assert will fire.
   static bool canInlineApplySite(FullApplySite apply);
 
+  /// Returns true if inlining \arg apply can result in improperly nested stack
+  /// allocations.
+  ///
+  /// In this case stack nesting must be corrected after inlining with the
+  /// StackNesting utility.
+  static bool needUpdateStackNesting(FullApplySite apply) {
```
Comment (rjmccall) [inline include/swift/SILOptimizer/Utils/SILInliner.h]: Usually predicates are written in the third-person singular present, so I think this should be need**s**UpdateStackNesting.

**Review by rjmccall — APPROVED** [review]
Approved with a minor request.

**Comment by eeckstein** [comment]
@swift-ci smoke test and merge

**Comment by eeckstein** [comment]
@swift-ci smoke test and merge

**Pull request merged**